pkg/github/tools.go

```diff
@@ -14,14 +16,126 @@ import (
 type GetClientFn func(context.Context) (*github.Client, error)
 type GetGQLClientFn func(context.Context) (*githubv4.Client, error)
 
-var DefaultTools = []string{"all"}
+// ToolsetMetadata holds metadata for a toolset including its ID and description
+type ToolsetMetadata struct {
+	ID          string
+	Description string
+}
+
+var (
+	ToolsetMetadataContext = ToolsetMetadata{
+		ID:          "context",
+		Description: "Tools that provide context about the current user and GitHub context you are operating in",
+	}
+	ToolsetMetadataRepos = ToolsetMetadata{
+		ID:          "repos",
+		Description: "GitHub Repository related tools",
+	}
+	ToolsetMetadataIssues = ToolsetMetadata{
+		ID:          "issues",
+		Description: "GitHub Issues related tools",
+	}
+	ToolsetMetadataPullRequests = ToolsetMetadata{
+		ID:          "pull_requests",
+		Description: "GitHub Pull Request related tools",
+	}
+	ToolsetMetadataUsers = ToolsetMetadata{
+		ID:          "users",
+		Description: "GitHub User related tools",
+	}
+	ToolsetMetadataOrgs = ToolsetMetadata{
+		ID:          "orgs",
+		Description: "GitHub Organization related tools",
+	}
+	ToolsetMetadataActions = ToolsetMetadata{
+		ID:          "actions",
+		Description: "GitHub Actions workflows and CI/CD operations",
+	}
+	ToolsetMetadataCodeSecurity = ToolsetMetadata{
+		ID:          "code_security",
+		Description: "Code security related tools, such as GitHub Code Scanning",
+	}
+	ToolsetMetadataSecretProtection = ToolsetMetadata{
+		ID:          "secret_protection",
+		Description: "Secret protection related tools, such as GitHub Secret Scanning",
+	}
+	ToolsetMetadataDependabot = ToolsetMetadata{
+		ID:          "dependabot",
+		Description: "Dependabot tools",
+	}
+	ToolsetMetadataNotifications = ToolsetMetadata{
+		ID:          "notifications",
+		Description: "GitHub Notifications related tools",
+	}
+	ToolsetMetadataExperiments = ToolsetMetadata{
+		ID:          "experiments",
+		Description: "Experimental features that are not considered stable yet",
+	}
+	ToolsetMetadataDiscussions = ToolsetMetadata{
+		ID:          "discussions",
+		Description: "GitHub Discussions related tools",
+	}
+	ToolsetMetadataGists = ToolsetMetadata{
+		ID:          "gists",
+		Description: "GitHub Gist related tools",
+	}
+	ToolsetMetadataSecurityAdvisories = ToolsetMetadata{
+		ID:          "security_advisories",
+		Description: "Security advisories related tools",
+	}
+	ToolsetMetadataProjects = ToolsetMetadata{
+		ID:          "projects",
+		Description: "GitHub Projects related tools",
+	}
+	ToolsetMetadataStargazers = ToolsetMetadata{
+		ID:          "stargazers",
+		Description: "GitHub Stargazers related tools",
+	}
+	ToolsetMetadataDynamic = ToolsetMetadata{
+		ID:          "dynamic",
+		Description: "Discover GitHub MCP tools that can help achieve tasks by enabling additional sets of tools, you can control the enablement of any toolset to access its tools when this toolset is enabled.",
```

Comment:
shall we include this in the default? This would allow agent to see what tools are available and prompt to enable them.